### ISSUE TITLE
[BUG] FNPR fails to update probability when move is rejected

### ISSUE THREAD

**Issue by bredelings** [revbayes/revbayes#576]
I tried running the revbayes tests on `rb --setOption debugMCMC=1` and it mostly worked.  One exception was the fossil-tip-age-uncertainty test.  One of the proposed moves set the `fbd_tree` probability to zero (-Inf), and after the move was rejected the correct probability was not restored.

To run the test I did `rb --setOption debugMCMC=1 scripts/FBDP_fossil_tip_age_uncetrainty.Rev --setOption logMCMC=3`, using rb from the `debug-mcmc` branch.

RevBayes complains that the `fbd_tree` probability is `-inf`, but when we touch it and recalculate it changes to `-167.367...`:
```
generation = 7    proposal = 60/270    NodeTimeSlideUniform(fbd_tree)
    BEFORE:   F[10]:  0.07257069283
    BEFORE:   phyMorpho:  -545.4601224
    BEFORE:   F[5]:  -1.568615918
    BEFORE:   F[11]:  -0.8415671857
    BEFORE:   F[6]:  -1.568615918
    BEFORE:   F[2]:  -0.8754687374
    BEFORE:   phySeq:  -4086.488094
    BEFORE:   F[4]:  -2.208274414
    BEFORE:   F[1]:  -1.047318994
    BEFORE:   F[14]:  -1.029619417
    BEFORE:   F[12]:  -1.252762968
    BEFORE:   F[7]:  0.05129329439
    BEFORE:   F[3]:  -0.9462378082
    BEFORE:   fbd_tree:  -inf
    BEFORE:   F[9]:  -1.193922468
    BEFORE:   F[13]:  1.500583508
    BEFORE:   F[8]:  0.5108256238

   Error:	Executing NodeTimeSlideUniform(fbd_tree)): PDFs not up-to-date before proposal!
       fbd_tree: -inf != -167.36752747    diff = -inf
```

But the real problem occurs earlier, when the probability is incorrectly set to `-Inf`:

```
generation = 7    proposal = 52/270    FNPR(fbd_tree)
    BEFORE:   F[10]:  0.07257069283
    BEFORE:   phyMorpho:  -545.4601224
    BEFORE:   F[5]:  -1.568615918
    BEFORE:   F[11]:  -0.8415671857
    BEFORE:   F[6]:  -1.568615918
    BEFORE:   F[2]:  -0.8754687374
    BEFORE:   phySeq:  -4086.454272
    BEFORE:   F[4]:  -2.208274414
    BEFORE:   F[1]:  -1.047318994
    BEFORE:   F[14]:  -1.029619417
    BEFORE:   F[12]:  -1.252762968
    BEFORE:   F[7]:  0.05129329439
    BEFORE:   F[3]:  -0.9462378082
    BEFORE:   fbd_tree:  -167.3675275
    BEFORE:   F[9]:  -1.193922468
    BEFORE:   F[13]:  1.500583508
    BEFORE:   F[8]:  0.5108256238

    PROPOSED: F[10]:  0.07257069283
    PROPOSED: phyMorpho:  -545.1993486
    PROPOSED: F[5]:  -1.568615918
    PROPOSED: F[11]:  -0.8415671857
    PROPOSED: F[6]:  -1.568615918
    PROPOSED: F[2]:  -0.8754687374
    PROPOSED: phySeq:  -4077.232065
    PROPOSED: F[4]:  -2.208274414
    PROPOSED: F[1]:  -1.047318994
    PROPOSED: F[14]:  -1.029619417
    PROPOSED: F[12]:  -1.252762968
    PROPOSED: F[7]:  0.05129329439
    PROPOSED: F[3]:  -0.9462378082
    PROPOSED: fbd_tree:  -inf
    PROPOSED: F[9]:  -1.193922468
    PROPOSED: F[13]:  1.500583508
    PROPOSED: F[8]:  0.5108256238

    FINAL:    F[10]:  0.07257069283
    FINAL:    phyMorpho:  -545.4601224
    FINAL:    F[5]:  -1.568615918
    FINAL:    F[11]:  -0.8415671857
    FINAL:    F[6]:  -1.568615918
    FINAL:    F[2]:  -0.8754687374
    FINAL:    phySeq:  -4086.454272
    FINAL:    F[4]:  -2.208274414
    FINAL:    F[1]:  -1.047318994
    FINAL:    F[14]:  -1.029619417
    FINAL:    F[12]:  -1.252762968
    FINAL:    F[7]:  0.05129329439
    FINAL:    F[3]:  -0.9462378082
    FINAL:    fbd_tree:  -inf
    FINAL:    F[9]:  -1.193922468
    FINAL:    F[13]:  1.500583508
    FINAL:    F[8]:  0.5108256238

    log(posterior_ratio) = -nan  log(likelihood_ratio) = 0   log(prior_ratio) = -nan
    log(acceptance_ratio) = -nan  log(hastings_ratio) = 0
  The move was REJECTED.
```

It also seems like the `log(prior_ratio)` should actually be `-inf`, not `nan`.  Possibly this isn't a problem with the FNPR move itself, but a problem in the Metropolis-Hastings code when the proposed state has probability `0.0`?

**Comment by davidcerny** [comment]
I can no longer reproduce this on the latest version of development ([3f80de7](https://github.com/revbayes/revbayes/commit/3f80de7074e37848644a9c7624f84d8bf249d5d6)). Is it possible that the problem had to do with `mvRootTimeSlideUniform` rather than `mvFNPR`, and that you fixed it with #594 @bredelings?

**Comment by bredelings** [comment]
I think so.

Fixed by #594
